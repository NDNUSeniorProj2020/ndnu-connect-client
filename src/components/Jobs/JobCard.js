import React from 'react';
import PropTypes from 'prop-types';
import { Card } from 'antd';
import moment from 'moment';

export default function JobCard({ job }) {
	if (job)
		return (
			<div>
				<Card title={job.title} extra={<a href={job.link}>Apply</a>}>
					<p>{job.description}</p>
					<p>{moment(job.date).format('LLLL')}</p>
				</Card>
			</div>
		);

	return null;
}

JobCard.propTypes = { job: PropTypes.object };
JobCard.defaultProps = { 
	job: {
		id: 0,
		title: '',
		description: '',
		qualifications: '',
		pay: '',
		link: '',
		date: '',
		type: 'FULL',
		user: 0
	}
};
