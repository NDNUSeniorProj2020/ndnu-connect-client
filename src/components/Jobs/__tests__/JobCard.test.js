import React from 'react';
import { shallow, mount } from 'enzyme';

import JobCard from '../JobCard';

describe('tests for JobCard', () => {
	describe('snapshot tests', () => {
		it('renders without crashing', () => {
			const tree = shallow(<JobCard />);
			expect(tree.html()).toMatchSnapshot();
		});
	});

	describe('unit tests', () => {
		it('renders default job if job prop is empty', () => {
			const defaultJob = {
				id: 0,
				title: '',
				description: '',
				qualifications: '',
				pay: '',
				link: '',
				date: '',
				type: 'FULL',
				user: 0
			};
			const wrapper = mount(<JobCard />);

			expect(wrapper.props().job).toEqual({ ...defaultJob });
		});
	});
});
