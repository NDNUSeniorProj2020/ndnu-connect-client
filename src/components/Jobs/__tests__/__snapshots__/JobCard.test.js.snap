// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`tests for JobCard snapshot tests renders without crashing 1`] = `"<div><div class=\\"ant-card ant-card-bordered\\"><div class=\\"ant-card-head\\"><div class=\\"ant-card-head-wrapper\\"><div class=\\"ant-card-extra\\"><a href=\\"\\">Apply</a></div></div></div><div class=\\"ant-card-body\\"><p></p><p>Invalid date</p></div></div></div>"`;
